Sming/Components/ssl/BearSsl/BrClientConnection.cpp

```diff
@@ -35,10 +35,11 @@ int BrClientConnection::init()
 	br_ssl_client_set_default_rsapub(&clientContext);
 
 	// X509 verification
-	x509Context = new X509Context([this]() { return context.getSession().validateCertificate(); });
-	br_ssl_engine_set_x509(getEngine(), *x509Context);
+	delete certificate;
+	certificate = new BrCertificate([this]() { return context.session.validateCertificate(); });
```

Comment:
could use `std::unique_ptr` instead.

Reply:
Could probably use that in a lot of places in the framework... I'm not sure about starting now...

Reply:
Please not here and not now. This is definitely work that should be organized in a separate PR. If we decide to go for std:unique_ptr, std::shared_ptr and family we should apply those change to the complete Sming Library (Core + Core Components + Sming only libraries).

Reply:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#rsmart-smart-pointers